wikimedia/wikipedia-ios
# Native Editor - Add heading support

**Phabricator:** 
https://phabricator.wikimedia.org/T348073

### Notes
Implements syntax highlighting and actions for heading buttons.

Notes:
- This PR will likely need to be reworked after #4696 is merged, but it can be merged before that one if needed.

### Test Steps
1. On Staging app, edit an article via the top navigation bar Edit button
2. Scroll to a header. Confirm headings syntax highlight as expected.
3. Edit the number of equal signs on an editing. Confirm font size grows and shrinks as expected.
4. Change font size in popover. Confirm font size grows and shrinks on headings in relation to other text as expected.
5. Change theme in popover. Confirm heading colors change as expected.
6. Toggle off syntax highlighting. Confirm syntax highlighting toggles as expected.
7. Select heading text. Open header formatting menu and confirm it shows a checkmark on the appropriate heading.
8. Select some of the other headings and confirm the appropriate equal signs appear or disappear.
9. Highlight a random word in a paragraph. Confirm it adds equal signs and new lines.
10. On a new empty line, tap a heading button. Confirm it adds the appropriate equal signs around an empty space.
